x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/xy_chart_plugin.tsx

```diff
@@ -141,84 +168,178 @@ const displayTypeIcon: { [type: string]: IconType } = {
+  const formattedNameX = xAxis
+    .map(op => operationToName(op.operation) + ('argument' in op ? ` of ${op.argument.field}` : ''))
+    .join(' split by ');
+  const formattedNameY = yAxis
+    .map(op => operationToName(op.operation) + ('argument' in op ? ` of ${op.argument.field}` : ''))
+    .join(' split by ');
+
   return {
-    previewExpression: toExpression(prefilledVisModel, 'preview'),
-    score: 0.5,
-    visModel: prefilledVisModel,
-    title,
-    iconType: displayTypeIcon[displayType],
-    pluginName: PLUGIN_NAME,
-    category: 'XY Chart',
+    ...visModel,
+    queries: {
+      xyChartQuery: {
+        datasourceRef: visModel.datasource!.title,
+        // Split by Y values, then bucket by X
+        select: yAxis.concat(xAxis),
+      },
+    },
+    editorPlugin: PLUGIN_NAME,
+    private: {
+      ...visModel.private,
+      xyChart: {
+        ...visModel.private.xyChart,
+        xAxis: {
+          title: formattedNameX,
+          columns: xAxis.map((_, index) => `xyChartQuery_${index + yAxis.length}`),
+        },
+        yAxis: {
+          title: formattedNameY,
+          columns: yAxis.map((_, index) => `xyChartQuery_${index}`),
+        },
+      },
+    },
   };
 }
 
+function _getSuggestionsForFieldAsReplacement(
+  field: DatasourceField,
+  visModel: XyChartVisModel
+): Suggestion[] {
+  const { datasource } = visModel;
+
+  if (!datasource) {
+    return [];
+  }
+
+  let suggestions = [] as Array<Suggestion | null>;
+
+  const opToSuggestion = (op: SelectOperator): Suggestion | null => {
+    let xAxis = [];
+    let yAxis = [];
+
+    if (op === 'count') {
+      return null;
+    }
+
+    xAxis = [fieldToOperation(field, op)];
+    yAxis = [fieldToOperation(field, 'count')];
+
+    const newVisModel = buildViewModel(visModel, xAxis, yAxis);
+
+    return buildSuggestion(newVisModel, {
+      iconType: displayTypeIcon.line,
+    });
+  };
+
+  const opWithDateHistogram = (op: SelectOperator): Suggestion | null => {
+    let xAxis = [];
+    let yAxis = [];
+
+    if (op === 'column') {
+      xAxis = [
+        fieldToOperation(
+          datasource.fields.find(f => f.name === datasource.timeFieldName)!,
+          'column'
+        ),
+      ];
+      yAxis = [fieldToOperation(field, op)];
+    } else {
+      xAxis = [
+        fieldToOperation(
+          datasource.fields.find(f => f.name === datasource.timeFieldName)!,
+          'date_histogram'
+        ),
+      ];
+
+      if (op === 'count') {
+        return null;
+      } else {
+        yAxis = [fieldToOperation(field, op), fieldToOperation(field, 'count')];
+      }
+    }
+
+    const newVisModel = buildViewModel(visModel, xAxis, yAxis);
+
+    return buildSuggestion(newVisModel, {
+      iconType: displayTypeIcon.line,
+    });
+  };
+
+  if (datasource!.timeFieldName && datasource!.timeFieldName !== field.name) {
+    if (field.type === 'number') {
+      suggestions = suggestions.concat(opWithDateHistogram('column'));
```

Comment:
For raw data in a line chart we need to order the x-axis
<img width="368" alt="Screenshot 2019-04-05 at 11 38 41" src="https://user-images.githubusercontent.com/1508364/55618687-6bb1bf80-5797-11e9-9220-88fae490f01e.png">
But that's probably something which should be done in a separate PR

Is there a specific reason why are we only suggesting the raw document version for number fields?

Reply:
There is a restriction that the Y axis in elastic-charts must be numeric, but the X axis might not have to be. I haven't tried it